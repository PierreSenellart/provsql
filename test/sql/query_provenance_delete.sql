\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE delete_test(id INT PRIMARY KEY);

INSERT INTO delete_test (id)
VALUES (1), (2), (3), (4), (5);

SELECT add_provenance('delete_test');

-- Test 1: single row deletion
SELECT create_provenance_mapping('delete_test_id', 'delete_test', 'id');
SELECT COUNT(*) FROM delete_test_id;
SELECT probability_evaluate(provenance) FROM delete_test_id WHERE value = 1;

DROP TABLE delete_test_id;
DELETE FROM delete_test WHERE id = 1;
SELECT create_provenance_mapping('delete_test_id', 'delete_test', 'id');
SELECT COUNT(*) FROM delete_test_id;
SELECT probability_evaluate(provenance) FROM delete_test_id WHERE value = 1;

-- Test 2: multiple rows deletion
DROP TABLE delete_test_id;
DELETE FROM delete_test WHERE id = 1;
SELECT create_provenance_mapping('delete_test_id', 'delete_test', 'id');
SELECT COUNT(*) FROM delete_test_id;
SELECT probability_evaluate(provenance) FROM delete_test_id WHERE value = 1;

DROP TABLE delete_test_id;
DELETE FROM delete_test WHERE id >= 2 AND id <= 4;
SELECT create_provenance_mapping('delete_test_id', 'delete_test', 'id');
SELECT COUNT(*) FROM delete_test_id;
SELECT value, probability_evaluate(provenance) FROM delete_test_id ORDER BY value;

-- Test 3: delete token tracking/ logging into delete_provenance table
SELECT query FROM delete_provenance ORDER BY deleted_at;

DROP TABLE delete_test;
DROP TABLE delete_test_id;
