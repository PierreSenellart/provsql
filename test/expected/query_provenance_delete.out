\set ECHO none
add_provenance

(1 row)
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
1
(1 row)
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
0
(1 row)
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
0
(1 row)
create_provenance_mapping

(1 row)
count
5
(1 row)
value|probability_evaluate
1|0
2|0
3|0
4|0
5|1
(5 rows)
query
DELETE FROM delete_test WHERE id = 1;
DELETE FROM delete_test WHERE id = 1;
DELETE FROM delete_test WHERE id >= 2 AND id <= 4;
(3 rows)
